lib/chef/knife/ec2_server_create.rb

```diff
@@ -824,7 +824,7 @@ def create_server_def
           :groups => config[:security_groups],
           :security_group_ids => locate_config_value(:security_group_ids),
           :flavor_id => locate_config_value(:flavor),
-          :key_name => Chef::Config[:knife][:aws_ssh_key_id],
+          :key_name => config[:ssh_key_name] || Chef::Config[:knife][:aws_ssh_key_id],
```

Comment:
I think we can replace this with `locate_config_value(:ssh_key_name)` . locate_config_value will take care of precedence.